sw/airborne/modules/system_identification/random.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) Joost Meulenbeld
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/system_identification/random.c"
+ * @author Joost Meulenbeld
+ * Random number functions
+ */
+
+#include "random.h"
+
+void init_random(void)
+{
+  srand(get_sys_time_msec());
+}
+
+double rand_uniform(void)
+{
```

Comment:
We typically use floats for airborne code. Also, this file doens't look well formatted.

This seems more appropriate in the math folder, then it might be used by other modules.

Reply:
I remember that on the original source of this code it said that for making sure the numbers are random enough it should be double, I think. If for some platforms this is a problem, they can set the CHIRP_USE_NOISE flag to false. Do you agree with this approach?

I also agree on putting it in the math folder. There are more modules that use this kind of code so they may switch to using a single implementation for it. But maybe that should be a separate pull request, what do you think?